AddOppTarget.cmake

```diff
@@ -52,15 +53,17 @@ function(add_opp_target)
         get_filename_component(msg_name "${msg_file}" NAME_WE)
         get_filename_component(msg_dir "${msg_file}" DIRECTORY)
         file(RELATIVE_PATH msg_prefix ${args_SOURCE_DIR} ${msg_dir})
-        set(msg_out_dir ${msg_gen_dir}/${msg_prefix})
-        file(MAKE_DIRECTORY ${msg_out_dir})
-        add_custom_command(OUTPUT "${msg_out_dir}/${msg_name}_m.cc" "${msg_out_dir}/${msg_name}_m.h"
-            COMMAND ${OMNETPP_MSGC} ARGS -s _m.cc -h ${msg_file}
-            DEPENDS ${msg_file} ${OMNETPP_MSGC}
-            COMMENT "Generating message ${msg_prefix}/${msg_name} of ${args_TARGET}"
-            WORKING_DIRECTORY ${msg_out_dir} VERBATIM)
 
-        list(APPEND cpp_files "${msg_out_dir}/${msg_name}_m.cc" "${msg_out_dir}/${msg_name}_m.h")
+        generate_opp_message(
+            ${msg_file}
+            OUTPUT_ROOT             ${msg_gen_dir}
+            DIRECTORY               ${msg_prefix}
+            SRCS                    _cpp_files
+            ADDITIONAL_NED_PATHS    ${args_SOURCE_DIR}
+            MSG4
```

Comment:
You define *MSG4* as an option to `add_opp_target` but always set it here no matter what.

Reply:
yep, will fix that